scipy/stats/_continuous_distns.py

```diff
@@ -782,12 +788,41 @@ def _pdf(self, x, c, d):
         # burr.pdf(x, c, d) = c * d * x**(-c-1) * (1+x**(-c))**(-d-1)
         return c * d * (x**(-c - 1.0)) * ((1 + x**(-c))**(-d - 1.0))
 
+    def _logpdf(self, x, c, d):
+        return (np.log(c) + np.log(d) + (-c - 1.0) * np.log(x) +
+                 (-d - 1.0) * np.log1p(x**(-c)))
+
     def _cdf(self, x, c, d):
         return (1 + x**(-c))**(-d)
 
+    def _logcdf(self, x, c, d):
+        return -d * np.log1p(x**(-c))
+
+    def _sf(self, x, c, d):
+        return 1 - (1 + x**(-c))**(-d)
```

Comment:
Maybe add _logsf using log1p, too?

Reply:
yes. Will switch `return 1 - (1 + x**(-c))**(-d)` to `return np.exp(self._logsf(x,c,d))`